### PR TITLE
Automated cherry pick of #99839: Cleanup portforward streams after their usage

### PR DESCRIPTION
**Pull request by saschagrunert**
Cherry pick of #99839 on release-1.21.

#99839: Cleanup portforward streams after their usage

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

### CODE REVIEW

**Comment by ehashman** [comment]
@saschagrunert the original fix is set as priority backlog, are we sure we want to backport this? Has this soaked enough to get signal either way?

**Comment by saschagrunert** [comment]
> Has this soaked enough to get signal either way?

I don't know, how can we decide that? Would you prefer in not cherry-picking this change and wait for some feedback after v1.22.0 got released?

**Review by ehashman — COMMENTED** [review]
SIG CLI tests which capture this look green.

/priority important-longterm
/triage accepted
/lgtm

**Comment by saschagrunert** [comment]
We still need approval on this one @lavalamp @deads2k @sttts

**Comment by saschagrunert** [comment]
@kubernetes/release-managers asking for approval.

**Review by hasheddan — APPROVED** [review]
/lgtm

@saschagrunert holding off on CP approval until backports to other branches are `approved` :+1:

**Pull request merged**
